comp/core/tagger/tagger.go

```diff
@@ -324,11 +327,48 @@ func (t *TaggerClient) ResetCaptureTagger() {
 // NOTE(remy): it is not needed to sort/dedup the tags anymore since after the
 // enrichment, the metric and its tags is sent to the context key generator, which
 // is taking care of deduping the tags while generating the context key.
-func (t *TaggerClient) EnrichTags(tb tagset.TagsAccumulator, udsOrigin string, clientOrigin string, cardinalityName string) {
-	cardinality := taggerCardinality(cardinalityName)
+func (t *TaggerClient) EnrichTags(tb tagset.TagsAccumulator, originInfo taggertypes.OriginInfo) {
```

Comment:
I think this function will need a solid set of unit tests to cover all the different cases

Reply:
Yes, I've added some more tests regarding the Opt-Out mechanism for now. Since we will be doing some future modifications, more tests will come in future PRs.